### PR TITLE
gh-117764: Fix and add signatures for many builtins

### PR DESCRIPTION
**Pull request by serhiy-storchaka**
* Issue: gh-117764

### CODE REVIEW

File: Objects/memoryobject.c [hunk 1]
```diff
@@ -3255,6 +3255,9 @@ PyDoc_STRVAR(memory_f_contiguous_doc,
              "A bool indicating whether the memory is Fortran contiguous.");
 PyDoc_STRVAR(memory_contiguous_doc,
              "A bool indicating whether the memory is contiguous.");
+PyDoc_STRVAR(memory_exit_doc,
+             "__exit__($self, /, *exc_info)\n--\n\n"
+             "Release the underlying buffer exposed by the memoryview object.");
```
Comment (eendebakpt) [inline Objects/memoryobject.c:3260]:
```suggestion
PyDoc_STRVAR(memory_enter_doc,
             "__enter__($self)\n--\n\n"
             "Check the underlying buffer has not been released.");
PyDoc_STRVAR(memory_exit_doc,
             "__exit__($self, /, *exc_info)\n--\n\n"
             "Release the underlying buffer exposed by the memoryview object.");
```
(maybe the signature for __enter__ is not correct, it returns a reference to itself)

File: Objects/memoryobject.c [hunk 2]
```diff
@@ -3283,7 +3286,7 @@ static PyMethodDef memory_methods[] = {
     MEMORYVIEW_TOREADONLY_METHODDEF
     MEMORYVIEW__FROM_FLAGS_METHODDEF
     {"__enter__",   memory_enter, METH_NOARGS, NULL},
```
Comment (eendebakpt) [inline Objects/memoryobject.c:3288]:
```suggestion
    {"__enter__",   memory_enter, METH_NOARGS, memory_enter_doc},
```

File: Objects/memoryobject.c (hunk 1 above)
Comment (serhiy-storchaka) [reply Objects/memoryobject.c:3260]: It is rather the side effect.

File: Objects/namespaceobject.c [hunk 3]
```diff
@@ -225,9 +225,9 @@ static PyMethodDef namespace_methods[] = {
 
 
 PyDoc_STRVAR(namespace_doc,
-"A simple attribute-based namespace.\n\
-\n\
-SimpleNamespace(**kwargs)");
+"SimpleNamespace(**kwargs)\n\
```
Comment (erlend-aasland) [inline Objects/namespaceobject.c:228]:
```suggestion
"SimpleNamespace($module, **kwargs)\n\
```

File: Objects/genericaliasobject.c [hunk 4]
```diff
@@ -537,6 +537,8 @@ _Py_subs_parameters(PyObject *self, PyObject *args, PyObject *parameters, PyObje
 }
 
 PyDoc_STRVAR(genericalias__doc__,
+"GenericAlias(origin, args, /)\n"
```
Comment (erlend-aasland) [inline Objects/genericaliasobject.c:540]:
```suggestion
"GenericAlias($module, origin, args, /)\n"
```

File: Objects/descrobject.c [hunk 5]
```diff
@@ -1165,8 +1165,8 @@ mappingproxy_reversed(PyObject *self, PyObject *Py_UNUSED(ignored))
 
 static PyMethodDef mappingproxy_methods[] = {
     {"get",       _PyCFunction_CAST(mappingproxy_get), METH_FASTCALL,
-     PyDoc_STR("D.get(k[,d]) -> D[k] if k in D, else d."
-               "  d defaults to None.")},
+     PyDoc_STR("get(self, key, default=None, /)\n--\n\n"
```
Comment (erlend-aasland) [inline Objects/descrobject.c]:
```suggestion
     PyDoc_STR("get($self, key, default=None, /)\n--\n\n"
```

File: Objects/namespaceobject.c (hunk 3 above)
Comment (serhiy-storchaka) [reply Objects/namespaceobject.c:228]: It is not a function.

File: Objects/memoryobject.c (hunk 1 above)
Comment (serhiy-storchaka) [reply Objects/memoryobject.c:3260]: I left this for future. The `__enter__` methods either acquire the resource, in which case it should be explicitly documented, or do not have a side effect, in which case they do not need a docstring. I am not sure what is closer to this case.

**Pull request merged**